--- origsrc/wxPython-src-3.0.2.0/build/aclocal/bakefile.m4	2013-12-16 07:42:29.000000000 -0600
+++ src/wxPython-src-3.0.2.0/build/aclocal/bakefile.m4	2015-01-04 14:53:55.788740800 -0600
@@ -639,7 +639,7 @@ AC_DEFUN([AC_BAKEFILE_CHECK_BASIC_STUFF]
 
     LDFLAGS_GUI=
     case ${BAKEFILE_HOST} in
-        *-*-cygwin* | *-*-mingw32* )
+        *-*-mingw32* )
         LDFLAGS_GUI="-mwindows"
     esac
     AC_SUBST(LDFLAGS_GUI)
--- origsrc/wxPython-src-3.0.2.0/include/wx/defs.h	2014-10-08 11:51:06.000000000 -0500
+++ src/wxPython-src-3.0.2.0/include/wx/defs.h	2017-03-28 21:43:23.887410400 -0500
@@ -256,7 +256,7 @@ typedef short int WXTYPE;
     #if defined(__VISUALC__) && (__VISUALC__ >= 1100)
         /*  VC++ 6.0 and 5.0 have explicit (what about earlier versions?) */
         #define HAVE_EXPLICIT
-    #elif ( defined(__MINGW32__) || defined(__CYGWIN32__) ) \
+    #elif ( defined(__MINGW32__) || defined(__CYGWIN__) ) \
           && wxCHECK_GCC_VERSION(2, 95)
         /*  GCC 2.95 has explicit, what about earlier versions? */
         #define HAVE_EXPLICIT
@@ -343,7 +343,7 @@ typedef short int WXTYPE;
         /*  VC++ 6.0 and 5.0 have std::string::compare */
         /*  (what about earlier versions?) */
         #define HAVE_STD_STRING_COMPARE
-    #elif ( defined(__MINGW32__) || defined(__CYGWIN32__) ) \
+    #elif ( defined(__MINGW32__) || defined(__CYGWIN__) ) \
           && wxCHECK_GCC_VERSION(3, 1)
         /*  GCC 3.1 has std::string::compare; */
         /*  3.0 never was in MinGW, 2.95 hasn't it */
@@ -438,7 +438,7 @@ typedef short int WXTYPE;
         Mingw <= 3.4 and all versions of Cygwin don't have std::wostream
      */
     #if (defined(__MINGW32__) && !wxCHECK_GCC_VERSION(4, 0)) || \
-        defined(__CYGWIN__)
+        (defined(__CYGWIN__) && !wxCHECK_GCC_VERSION(5, 0))
         #define wxNO_WOSTREAM
     #endif
 
--- origsrc/wxPython-src-3.0.2.0/src/common/mimecmn.cpp	2013-12-16 07:42:30.000000000 -0600
+++ src/wxPython-src-3.0.2.0/src/common/mimecmn.cpp	2017-03-28 21:48:12.630902100 -0500
@@ -596,7 +596,7 @@ bool wxMimeTypesManager::Unassociate(wxF
 {
     EnsureImpl();
 
-#if defined(__UNIX__) && !defined(__CYGWIN__) && !defined(__WINE__)
+#if defined(__UNIX__) && !defined(__WINE__)
     return m_impl->Unassociate(ft);
 #else
     return ft->Unassociate();
@@ -706,7 +706,7 @@ size_t wxMimeTypesManager::EnumAllFileTy
 void wxMimeTypesManager::Initialize(int mcapStyle,
                                     const wxString& sExtraDir)
 {
-#if defined(__UNIX__) && !defined(__CYGWIN__) && !defined(__WINE__)
+#if defined(__UNIX__) && !defined(__WINE__)
     EnsureImpl();
 
     m_impl->Initialize(mcapStyle, sExtraDir);
@@ -719,7 +719,7 @@ void wxMimeTypesManager::Initialize(int
 // and this function clears all the data from the manager
 void wxMimeTypesManager::ClearData()
 {
-#if defined(__UNIX__) && !defined(__CYGWIN__) && !defined(__WINE__)
+#if defined(__UNIX__) && !defined(__WINE__)
     EnsureImpl();
 
     m_impl->ClearData();
--- origsrc/wxPython-src-3.0.2.0/src/unix/stdpaths.cpp	2013-12-16 07:42:31.000000000 -0600
+++ src/wxPython-src-3.0.2.0/src/unix/stdpaths.cpp	2015-01-04 14:54:33.743607000 -0600
@@ -37,7 +37,7 @@
 #include "wx/log.h"
 #include "wx/textfile.h"
 
-#if defined( __LINUX__ ) || defined( __VMS )
+#if defined( __LINUX__ ) || defined( __VMS ) || defined ( __CYGWIN__ )
     #include <unistd.h>
 #endif
 
@@ -117,7 +117,7 @@ wxString wxStandardPaths::GetExecutableP
 
 wxString wxStandardPaths::GetExecutablePath() const
 {
-#ifdef __LINUX__
+#if defined ( __LINUX__ ) || defined ( __CYGWIN__ )
     wxString exeStr;
 
     char buf[4096];
--- origsrc/wxPython-src-3.0.2.0/src/unix/utilsunx.cpp	2014-06-24 16:43:01.000000000 -0500
+++ src/wxPython-src-3.0.2.0/src/unix/utilsunx.cpp	2015-01-04 14:56:24.894800900 -0600
@@ -1172,7 +1172,7 @@ unsigned long wxGetProcessId()
 
 wxMemorySize wxGetFreeMemory()
 {
-#if defined(__LINUX__)
+#if defined(__LINUX__) || defined(__CYGWIN__)
     // get it from /proc/meminfo
     FILE *fp = fopen("/proc/meminfo", "r");
     if ( fp )
