diff -up wxWidgets-3.0.4/tests/filename/filenametest.cpp.fix_fntest wxWidgets-3.0.4/tests/filename/filenametest.cpp
--- wxWidgets-3.0.4/tests/filename/filenametest.cpp.fix_fntest	2018-03-07 11:55:38.000000000 -0500
+++ wxWidgets-3.0.4/tests/filename/filenametest.cpp	2018-11-22 11:58:40.099232797 -0500
@@ -766,8 +766,7 @@ void FileNameTestCase::TestExists()
 #ifdef __LINUX__
     // These files are only guaranteed to exist under Linux.
     // No need for wxFILE_EXISTS_NO_FOLLOW here; wxFILE_EXISTS_SYMLINK implies it
-    CPPUNIT_ASSERT( wxFileName::Exists("/dev/core", wxFILE_EXISTS_SYMLINK) );
-    CPPUNIT_ASSERT( wxFileName::Exists("/dev/log", wxFILE_EXISTS_SOCKET) );
+    CPPUNIT_ASSERT( wxFileName::Exists("/proc/self", wxFILE_EXISTS_SYMLINK) );
 #endif // __LINUX__
 #ifndef __VMS
     wxString fifo = dirTemp.GetPath() + "/fifo";
